docs/docsite/rst/dev_guide/module_lifecycle.rst

```diff
@@ -0,0 +1,46 @@
+.. _module_lifecycle:
```

Comment:
Example of giving notice of deprecation for a set of modules https://github.com/ansible/ansible/pull/43781/files

Reply:
thanks, that's a better link than the kubernetes one